molpipeline/estimators/chemprop/abstract.py

```diff
@@ -95,3 +117,106 @@ def fit(
         )
         self.lightning_trainer.fit(self.model, training_data)
         return self
+
+    def set_params(self, **params: Any) -> Self:
+        """Set the parameters of the model.
+
+        Note
+        ----
+        Parameters for the trainer and the checkpoint callback are filtered out and added as attributes of the model.
+        This is done due to incompatibility with the `get_params` method.
+
+        Parameters
+        ----------
+        **params: Any
+            The parameters to set.
+
+        Returns
+        -------
+        Self
+            The model with the new parameters.
+        """
+        params, trainer_params = self._filter_params_trainer(params)
+        params, model_ckpoint_params = self._filter_params_callback(params)
+        self.trainer_params.update(trainer_params)
+        self.model_ckpoint_params.update(model_ckpoint_params)
+        super().set_params(**params)
```

Comment:
Do we want to check here wether a parameter which was already set to a non-default value would be updated? (i.e. if I pass an initialized trainer with max_epochs=50 and the also pass lightning_trainer__max_epochs=100 this will lead to max_epochs being set to 100 but I would say we should raise an error as two non-default values were passed?

Reply:
I think in sklearn the object is first set and after collecting all its parameters from the kwargs the object is updated accordingly. [See here](https://github.com/scikit-learn/scikit-learn/blob/e12f192c581ff78b25a6cac723a99782c9ee480d/sklearn/base.py#L285)
So for now I would leave it as is, which overwrites trainer params with potential kwargs,